test/ret_web/channels/auth_channel_test.exs

```diff
@@ -0,0 +1,51 @@
+defmodule RetWeb.AuthChannelTest do
+  use RetWeb.ChannelCase
+  import Ecto.Query
+
+  alias RetWeb.{SessionSocket}
+  alias Ret.{Account, AppConfig, Repo}
+
+  @test_email "admin1@mozilla.com"
+  @test_email2 "admin2@mozilla.com"
+
+  setup do
+    {:ok, socket} = connect(SessionSocket, %{})
+    {:ok, %{session_id: _session_id}, socket} = subscribe_and_join(socket, "auth:testauth", %{})
+    {:ok, socket: socket}
+  end
+
+  test "Login token is created for an email", %{socket: socket} do
+    refute login_token_for_email_exists?(@test_email)
+    refute Account.exists_for_email?(@test_email)
+
+    push(socket, "auth_request", %{"email" => @test_email, "origin" => "test"})
+    :timer.sleep(500)
```

Comment:
Is this sleep really necessary?

Reply:
Yep, unfortunately the channel messaging going on here is not request/response, we currently (perhaps erroneously) do not send a reply when the email is sent. I don't remember if there was an intentional reason to do this, but without a reply the only way to test the side effects of the message is to just wait long enough.